libtiledbsoma/src/stats.cc

```diff
@@ -0,0 +1,57 @@
+/**
+ * @file   stats.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file provides access to stats from libtiledbsoma's dependency on
+ * TileDB Embedded.
+ */
+
+#include "tiledbsoma/stats.h"
+#include <tiledb/tiledb>
+
+namespace tiledbsoma::stats {
+
+void enable() {
+    tiledb::Stats::enable();
+}
+
+void disable() {
+    tiledb::Stats::disable();
+}
+
+void reset() {
+    tiledb::Stats::reset();
+}
+
+std::string dump() {
+    std::string stats;
+    tiledb::Stats::raw_dump(&stats);
+    return stats;
```

Comment:
These still just pass through to TileDB Core methinks, libtiledbsoma has no separate stats instance.  So while you move the accessor into a different namespace I have not yet convinced myself actually get different stats.

Reply:
as in the description of this PR -- i've not yet debugged everything on #1066 but the pushdown of this code to `libtiledbsoma` is something that is worth doing